packages/comments/src/stores/comments/CommentsProvider.tsx

```diff
@@ -74,35 +66,83 @@ export const { commentsStore, useCommentsStore } = createAtomStore(
     newValue: [{ type: 'p', children: [{ text: '' }] }],
 
     focusTextarea: false,
-
-    onCommentAdd: null,
-    onCommentUpdate: null,
-    onCommentDelete: null,
-  } as CommentsStoreState,
+  } satisfies CommentsStoreState as CommentsStoreState,
   {
     name: 'comments',
     scope: SCOPE_COMMENTS,
   }
 );
 
+export const CommentsContext = createContext<CommentsContext>({
+  myUserId: null,
+  users: {},
+  onCommentAdd: null,
+  onCommentUpdate: null,
+  onCommentDelete: null,
+});
+
+export interface CommentsProviderProps extends Partial<CommentsContext> {
+  initialComments?: CommentsStoreState['comments'];
+  children: ReactNode;
+}
+
 export function CommentsProvider({
   children,
-  ...props
-}: Partial<CommentsStoreState> & { children: ReactNode }) {
+  initialComments,
+  myUserId = null,
+  users = {},
+  onCommentAdd = null,
+  onCommentUpdate = null,
+  onCommentDelete = null,
+}: CommentsProviderProps) {
   return (
     <JotaiProvider
-      initialValues={getJotaiProviderInitialValues(commentsStore, props)}
+      initialValues={getJotaiProviderInitialValues(commentsStore, {
+        comments: initialComments,
+      })}
       scope={SCOPE_COMMENTS}
     >
-      {children}
+      <CommentsContext.Provider
+        value={{
+          myUserId,
+          users,
+          onCommentAdd,
+          onCommentUpdate,
+          onCommentDelete,
+        }}
+      >
+        {children}
+      </CommentsContext.Provider>
     </JotaiProvider>
   );
 }
 
 export const useCommentsStates = () => useCommentsStore().use;
-export const useCommentsSelectors = () => useCommentsStore().get;
 export const useCommentsActions = () => useCommentsStore().set;
 
+export const useCommentsSelectors = () => {
+  const context = useContext(CommentsContext);
+
+  const contextGetters = useMemo(
+    () =>
+      Object.fromEntries(
+        Object.entries(context).map(([key, value]) => [key, () => value])
+      ) as { [K in keyof CommentsContext]: () => CommentsContext[K] },
+    [context]
+  );
+
+  const storeGetters = useCommentsStore().get;
+
+  // Combine getters from context and store
```

Comment:
If this combination of classic context and jōtai becomes a common pattern in the future, we could add a set of helpers for this.